sdk/python/packages/flet-core/src/flet_core/page.py

```diff
@@ -1000,311 +1009,251 @@ def __on_invoke_method_result(self, e) -> None:
         self.__method_call_results[evt] = (result.result, result.error)
         evt.set()
 
-    # Open / Close
-    def __open_control(
-        self, control_name: str, control: Control, base: "Union[Page, Offstage]" = None
-    ) -> None:
-        if not base:
-            base = self
-
-        if control:
-            control.open = True
-            base.update()
+    def open(self, control: Control) -> None:
+        if not hasattr(control, "open"):
+            raise ValueError("control has no open attribute")
         else:
-            raise Exception(
-                f"Please define {control_name} before show_{control_name}() method"
-            )
-
-    def __close_control(
-        self, control_name: str, control: Control, base: "Union[Page, Offstage]" = None
-    ) -> None:
-        if not base:
-            base = self
+            control.open = True
+            if isinstance(control, NavigationDrawer):
+                if control.position == NavigationDrawerPosition.END:
+                    if self.end_drawer == control:
+                        control.update()
+                        return
+                    else:
+                        self.end_drawer = control
+                else:
+                    if self.drawer == control:
+                        control.update()
+                        return
+                    else:
+                        self.drawer = control
+                self.update()  # called only if the new drawer is different from the current one
+            elif control not in self.overlay:
+                self.__offstage.controls.append(control)
+                self.__offstage.update()
+            return
 
-        if control:
+    @staticmethod
+    def close(control: Control) -> None:
+        if hasattr(control, "open"):
             control.open = False
-            base.update()
+            control.update()
```

Comment:
Shall we remove control from `self.__offstage.controls`?

Reply:
We could eventually... 
Any reason for that?